torch/csrc/jit/frontend/ir_emitter.cpp

```diff
@@ -1706,84 +1750,181 @@ struct to_ir {
   }
 
   CondValue emitIsInstance(const Expr& obj, const Expr& classinfo) {
-    // turn (float, (int, tuple)) into a flat list of types and type kind
-    // category checks: tuple_check = true, types = {float, int}
-    struct GatheredTypes {
-      GatheredTypes(ScriptTypeParser parser) : typeParser_(std::move(parser)) {}
-      void gather(const Expr& classinfo) {
-        if (classinfo.kind() == TK_TUPLE_LITERAL) {
-          for (Expr e : TupleLiteral(classinfo).inputs()) {
-            gather(e);
+    // For use with `unifyTypeList`
+    std::stringstream nowhere;
+
+    Value* lhs_val;
+
+    // Get the types on the right-hand side of the `isinstance`
```

Comment:
What do you think about writing at the top of this function a description of the algorithm ? 

> we also want to decompose any `Union` types
this describes what we are doing, but it doesn't describe the motivation